src/provider/provider_tracking.c

```diff
@@ -586,17 +586,23 @@ static umf_result_t trackingOpenIpcHandle(void *provider, void *providerIpcData,
         (umf_tracking_memory_provider_t *)provider;
     umf_result_t ret = UMF_RESULT_SUCCESS;
 
+    if (p->hUpstream == NULL) {
```

Comment:
How it is possible? If it happens it indicates error in UMF code, right? If so, I think it should be just assert.

I agree that `&& p->hUpstream` checks below should be just removed.

Reply:
This is just a fix for coverity. Agree that this should never happen but I think it is better to add this code that add a false-positive

Reply:
IMHO if it is really a false-positive it is much better to ignore it in Coverity than add a redundant code.

Reply:
@bratpiorka I propose sth like that https://github.com/oneapi-src/unified-memory-framework/pull/420 instead and ignore this issue in Coverity.